services/web/server/src/simcore_service_webserver/projects/_handlers_project_comments.py

```diff
@@ -0,0 +1,250 @@
+            project_uuid=path_params.project_uuid,
+            offset=query_params.offset,
+            limit=query_params.limit,
+        )
+
+        page = Page[dict[str, Any]].parse_obj(
+            paginate_data(
+                chunk=project_comments,
+                request_url=request.url,
+                total=total_project_comments,
+                limit=query_params.limit,
+                offset=query_params.offset,
+            )
+        )
+        return web.Response(
+            text=page.json(**RESPONSE_MODEL_POLICY),
+            content_type=MIMETYPE_APPLICATION_JSON,
+        )
+    except ProjectNotFoundError as exc:
+        raise web.HTTPNotFound(
+            reason=f"Project {path_params.project_uuid} not found"
+        ) from exc
+
+
+@routes.put(
+    f"/{VTAG}/projects/{{project_uuid}}/comments/{{comment_id}}",
+    name="update_project_comment",
+)
+@login_required
+@permission_required("project.read")
+async def update_project_comment(request: web.Request):
+    req_ctx = RequestContext.parse_obj(request)
+    path_params = parse_request_path_parameters_as(
+        _ProjectCommentsWithCommentPathParams, request
+    )
+    body_params = await parse_request_body_as(_ProjectCommentsBodyParams, request)
+
+    try:
+        # ensure the project exists
+        await projects_api.get_project_for_user(
+            request.app,
+            project_uuid=f"{path_params.project_uuid}",
+            user_id=req_ctx.user_id,
+            include_state=False,
+        )
+
+        return await _api_project_comments.update_project_comment(
+            request=request,
+            comment_id=path_params.comment_id,
+            project_uuid=path_params.project_uuid,
+            contents=body_params.contents,
+        )
+    except ProjectNotFoundError as exc:
+        raise web.HTTPNotFound(
+            reason=f"Project {path_params.project_uuid} not found"
+        ) from exc
+
+
+@routes.delete(
+    f"/{VTAG}/projects/{{project_uuid}}/comments/{{comment_id}}",
+    name="delete_project_comment",
+)
+@login_required
+@permission_required("project.read")
+async def delete_project_comment(request: web.Request):
+    req_ctx = RequestContext.parse_obj(request)
+    path_params = parse_request_path_parameters_as(
+        _ProjectCommentsWithCommentPathParams, request
+    )
+
+    try:
+        # ensure the project exists
+        await projects_api.get_project_for_user(
+            request.app,
+            project_uuid=f"{path_params.project_uuid}",
+            user_id=req_ctx.user_id,
+            include_state=False,
+        )
+
+        await _api_project_comments.delete_project_comment(
+            request=request,
+            comment_id=path_params.comment_id,
+        )
+        raise web.HTTPNoContent(content_type=MIMETYPE_APPLICATION_JSON)
+    except ProjectNotFoundError as exc:
+        raise web.HTTPNotFound(
+            reason=f"Project {path_params.project_uuid} not found"
```

Comment:
Note that `ProjectNotFoundError` has already a formatted message for this, therefore handling errors would simply be transforming`ProjectNotFoundError` into `HTTPNotFound` as 
```python
raise web.HTTPNotFound( reason=f"{exc}")
```

Reply:
Moreover, note that all these entrypoints convert `ProjectNotFoundError` -> `web.HTTPNotFound`. 
I highly recommend you to create a decorator to handler errors as

https://github.com/ITISFoundation/osparc-simcore/blob/0424ad5339a9caab8d73eea201b355b400ca34f4/services/web/server/src/simcore_service_webserver/projects/_handlers_project_nodes.py

and use it in every handler as

```python

@routes.delete(
    f"/{VTAG}/projects/{{project_uuid}}/comments/{{comment_id}}",
    name="delete_project_comment",
)
@login_required
@permission_required("project.read")
@_handle_project_errors # <---------
async def delete_project_comment(request: web.Request):
   ...
```

Reply:
Cool thanks! I have added it there. These reviews even though they take sometimes a lot of time are so useful! :+1: